mlir-clang/Lib/ValueCategory.cc

```diff
@@ -13,8 +13,23 @@
 #include "mlir/IR/OpDefinition.h"
 #include "mlir/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.h"
 #include "polygeist/Ops.h"
+#include <mlir/Dialect/Arithmetic/IR/Arithmetic.h>
 
 using namespace mlir;
+using namespace mlir::arith;
+
+ValueCategory::ValueCategory(mlir::Value val, bool isReference)
+    : val(val), isReference(isReference) {
+  assert(val);
```

Comment:
Any reason why this is back in?

Reply:
Yes, it is illegal to have a reference that isn't a pointer/memref (which was generated by accident earlier, and this allows it to be much easier to find)

Reply:
Ok! Thanks.